Sources/Charcoal/CharcoalViewController.swift

```diff
@@ -8,6 +8,8 @@ public protocol CharcoalViewControllerDelegate: class {
     func charcoalViewController(_ viewController: CharcoalViewController, didSelect vertical: Vertical)
     func charcoalViewController(_ viewController: CharcoalViewController, didSelectExternalFilterWithKey key: String, value: String?)
     func charcoalViewControllerDidPressShowResults(_ viewController: CharcoalViewController)
+    func charcoalViewControllerWillBeginTextEditing(_ viewController: CharcoalViewController)
+    func charcoalViewControllerWillEndTextEditing(_ viewController: CharcoalViewController)
     func charcoalViewController(_ viewController: CharcoalViewController,
```

Comment:
Maybe we can split this protocol to `CharcoalViewControllerSelectionDelegate` and `CharcoalViewControllerTextEditingDelegate`?

Reply:
Yes, we can do that. The same with `FilterViewControllerDelegate`? Have a separate protocol like `TextEditingDelegate` or something

Reply:
I think we can keep `FilterViewControllerDelegate` as long as it works for as because it's an internal protocol. But `CharcoalViewControllerDelegate` on the other hand is public, so it might be more flexible if we split it in two.

Reply:
Fixed